tools/find_device.c

```diff
@@ -0,0 +1,146 @@
+/*
+ * Copyright 2020 Codethink Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <dirent.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <libudev.h>
+#include <locale.h>
+#include <unistd.h>
+#include <stdbool.h>
+
+#include "find_device.h"
+
+static int match(char *sysname)
+{
+    struct udev *udev;
+    struct udev_enumerate *enumerate;
+    struct udev_list_entry *device;
+    struct udev_device *dev;
+    const char *path;
+    bool ret = false;
+
+    /* Create the udev object */
+    udev = udev_new();
+    if (!udev)
+    {
+        printf("Can't create udev\n");
+        exit(1);
+    }
+
+    /* Create a list of the devices with the name "ttyUSB*". */
+    enumerate = udev_enumerate_new(udev);
+    udev_enumerate_add_match_sysname(enumerate, sysname);
+    udev_enumerate_scan_devices(enumerate);
+
+    /* udev_enumerate_get_list_entry returns a list of entry,
+    but we only need the first entry, because we only cares
+    about their parent which are the same for all. */
+    device = udev_enumerate_get_list_entry(enumerate);
+
+    /* Get the filename of the /sys entry for the device
+     and create a udev_device object (dev) representing it */
+    path = udev_list_entry_get_name(device);
```

Comment:
Do we need a NULL check for path?

Reply:
I've not used udev much myself, so I'm not sure if we need error handling on all the other `udev_` calls here.

Reply:
there isn't much documentation for this library, or comments in the header file, but checking the source of it, yes, they need NULL checks.